### DISCUSSION TITLE
Prevent emitting enum if any case is the blank string

### DISCUSSION THREAD

**Discussion by Quafadas** [quicktype/quicktype#2227] [Q&A]
Is it possible to prevent emitting an enum type (can we go with string) if any of the cases discovered are the blank string?

**Comment by Quafadas** [comment]
@dvdsgl apologies for the mention - do you you happen to know if there is a way to direct QuickType away from enums if one of the cases is the blank string?

**Reply by dvdsgl** [reply]
No, but this could be an option. What's the use case?

**Reply by Quafadas** [reply]
In quicktypes test cases, it sometimes infers enums, when one of the cases is the blank string. Not necessarily incorrectly, in fairness. 

However, in scala / java, as far as I can tell, the blank string is not easily representable as an enum. My google fu / sketchy prompt engineering skillz have lead me to believe that workarounds (case EMPTY) to that quickly get complex and brittle, particulaly when it comes to parsing them.

To workaround this and "get something for scala over the line", I essentially made enums the union of a bunch of strings. i.e. 

`type blankableEnum = "" | "case1" | "case2"`

Which worked... but it has some drawbacks. An enum with many cases can be longer than a string the compiler will accept for example. 

Also, IDE support is more limited - no tab completion ... and the representation is, in general less intuitive. 

Although, after [writing this out](https://rubberduckdebugging.com/), I'm wondering if I couldn't bifurcate the existing implementation by checking the cases in advance of actually emitting something. 

- if blank is found, go with theexisting implementation 
- if no blank is found, emit a "proper enum"

So now actually, I'm not sure if it should be a quicktype feature. Hmmm... I will test this. Thanks for the response...
